src/libbson/src/bson/bson-atomic.c

```diff
@@ -17,79 +17,95 @@
 
 #include "bson-atomic.h"
 
+#if _POSIX_SOURCE
+/* For sched_yield() */
+#include <sched.h>
+#endif
 
-/*
- * We should only ever hit these on non-Windows systems, for which we require
- * pthread support. Therefore, we will avoid making a threading portability
- * for threads here and just use pthreads directly.
- */
+int32_t
+bson_atomic_int_add (volatile int32_t *p, int32_t n)
+{
+   return 1 + bson_atomic_int32_fetch_add (p, n, bson_memorder_seqcst);
+}
 
+int64_t
+bson_atomic_int64_add (volatile int64_t *p, int64_t n)
+{
+   return 1 + bson_atomic_int64_fetch_add (p, n, bson_memorder_seqcst);
+}
 
-#ifdef __BSON_NEED_BARRIER
-#include <pthread.h>
-static pthread_mutex_t gBarrier = PTHREAD_MUTEX_INITIALIZER;
 void
-bson_memory_barrier (void)
+bson_yield_thread ()
 {
-   pthread_mutex_lock (&gBarrier);
-   pthread_mutex_unlock (&gBarrier);
+   BSON_IF_WINDOWS (SwitchToThread ();)
```

Comment:
This appears to error on macOS build: https://spruce.mongodb.com/task/mongo_c_driver_releng_link_with_bson_mac_patch_d8684bed2f5d0763aceb8c03cbce1e2637bc515c_610adb7e2fbabe37ed81f877_21_08_04_18_25_04/logs?execution=0